calico_kubernetes/tests/kube_plugin_test.py

```diff
@@ -416,42 +416,45 @@ def test_delete_docker_interface(self):
             ], any_order=True)
 
     def test_configure_profile(self):
-        with patch_object(self.plugin, '_datastore_client',
-                          autospec=True) as m_datastore_client, \
-                patch_object(self.plugin, '_get_namespace_tag',
-                             autospec=True) as m_get_namespace_tag, \
-                patch_object(self.plugin, '_get_pod_config',
-                             autospec=True) as m_get_pod_config, \
-                patch_object(self.plugin, '_apply_rules',
-                             autospec=True) as m_apply_rules, \
-                patch_object(self.plugin, '_apply_tags',
-                             autospec=True) as m_apply_tags:
-            # Set up mock objects
-            m_datastore_client.profile_exists.return_value = False
-            m_get_pod_config.return_value = 'pod'
-            m_get_namespace_tag.return_value = 'tag'
-
-            # Set up class members
-            pod_name = 'pod_name'
-            profile_name = 'profile_name'
-            self.plugin.pod_name = pod_name
-            self.plugin.profile_name = profile_name
-
-            # Set up args
-            endpoint = Endpoint(TEST_HOST, TEST_ORCH_ID, '1234', '5678',
-                                'active', 'mac')
+        # Set up mock objects
+        m_datastore_client = Mock()
```

Comment:
Good spot.  We can't use `patch.object` as a decorator on `self.plugin` since `self` is not accessible in the decorator.  I think using `patch.object` is marginally safer, but you need to remember to spec anyway (either autospec, or explicitly) so I think I favor using MagicMock() to save the indentation.